rules/cross-platform/discovery_kernel_module_enumeration.toml

```diff
@@ -0,0 +1,47 @@
+[metadata]
+creation_date = "2020/04/23"
+ecs_version = ["1.5.0"]
+maturity = "production"
+updated_date = "2020/08/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+Loadable Kernel Modules (or LKMs) and Kernel Extensions are pieces of code that can be loaded and unloaded into the
+kernel upon demand. They extend the functionality of the kernel without the need to reboot the system. This identifies
+attempts to enumerate information about a kernel module or extension.
+"""
+false_positives = [
+    """
+    Security tools and device drivers may run these programs in order to enumerate kernel modules or extensions. Use of
+    these programs by ordinary users is uncommon. These can be exempted by process name or username.
+    """,
+]
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Enumeration of Kernel Modules or Extensions"
+note = "For macOS systems, Auditbeat must be run with an account that has access to the binaries in the /usr/sbin, /sbin, /usr/bin, and /bin directories. For Linux systems, to avoid conflicts with auditd, you will need to stop and disable auditd with 'sudo service stop auditd && sudo systemctl disable auditd'."
+risk_score = 47
+rule_id = "2d8043ed-5bda-4caf-801c-c1feb7410504"
+severity = "medium"
+tags = ["Elastic", "Linux", "macOS"]
+type = "query"
+
+query = '''
+event.category:process and event.type:start and process.args:((kmod and list) or (depmod or lsmod or modinfo or kextstat))
```

Comment:
Should this be separated into `process.name` and `process.args`?

Reply:
If you run the command as `root` or `sudo`, Auditbeat displays different information in `process.name`, but the binary is always in the `process.args`.

![image](https://user-images.githubusercontent.com/7442091/91188674-30c76900-e6b7-11ea-89b5-9b4e38fb5b49.png)

Looking again, I'm going to add `kmod and static-nodes`, they're the only two commands for the `kmod` binary.

```
kmod
missing command
kmod - Manage kernel modules: list, load, unload, etc
Usage:
	kmod [options] command [command_options]

Options:
	-V, --version     show version
	-h, --help        show this help

Commands:
  help         Show help message
  list         list currently loaded modules
  static-nodes outputs the static-node information installed with the currently running kernel

kmod also handles gracefully if called from following symlinks:
  lsmod        compat lsmod command
  rmmod        compat rmmod command
  insmod       compat insmod command
  modinfo      compat modinfo command
  modprobe     compat modprobe command
  depmod       compat depmod command
```

Reply:
when you run a command with sudo, (with the exception of internal bash commands), you should see _two_ process events when sudo is successful: the user running `sudo` + command line, then `root` running the intended command line

Reply:
could that be related to the issue you found the other day, when auditbeat is running with user, non-root privs?

Reply:
That is really odd...I wonder if because `tcpdump` is an external package vs. `kmod` (as well as the other binaries in this file) are all internal?

Here is running w/`sudo` and `root`...looks the same in the `process.name`.
![image](https://user-images.githubusercontent.com/7442091/91189832-881a0900-e6b8-11ea-8406-60708f402c0a.png)

Reply:
> could that be related to the issue you found the other day, when auditbeat is running with user, non-root privs?

This issue has boiled down to Auditbeat not using `auditd` for macOS. There are some directories that Auditbeat can't/doesn't/won't look in on macOS.